### PR TITLE
Allow HP admin to toggle HPOS features

### PR DESCRIPTION
**Pull request by filalex77**
Routes added to HP Admin:
- `GET /profiles`: get the whole TOML file generated by HP Admin
- `GET /profiles/<profile>/features/<feature>`: check if a feature is enabled
  - returns `{"enabled": bool}`
- `PUT /profiles/<profile>/features/<feature>`: enable a feature
- `DELETE /profiles/<profile>/features/<feature>`: disable a feature

### CODE REVIEW

**Comment by alastairong** [comment]
@filalex77 Has this been added in a way that means these options will be automatically enabled for every HoloPort **without** requiring any of our users to manually access `configuration.nix`?

**Comment by alastairong** [comment]
@filalex77 I also forgot that per our definition of done, you will need to update the RFC. Feel free to add it to this https://github.com/Holo-Host/rfcs/pull/49

And then we should merge that documentation.

**Comment by peeech** [comment]
Temporarily closing - my suspicion is that a kind of errors that this PR gives on Hydra is a source of our problems with hanging Hydra.

**Comment by filalex77** [comment]
Fixed infinite recursion errors in 0774300444b89aae318e222f9d54cd8fac37db83, and allowed for `/etc/nixos/hp-admin-features.toml` to not exist in 3c01df628b9d1c83ac8e30108895489399d10fd2.

Reopening this PR, as it should be safe for Hydra evaluation now.

**Comment by alastairong** [comment]
@filalex77 Fails to build. Looks like a syntax error in the tests

**Comment by alastairong** [comment]
@filalex77 I pushed up a fix for the perl syntax. Still failing though

File: overlays/holo-nixpkgs/hpos-admin/hpos-admin.py [hunk 1]
```diff
@@ -213,7 +217,7 @@ def status():
 
 @app.route('/upgrade', methods=['POST'])
 def upgrade():
-    rebuild(priority=1, args=['--upgrade'])
+    rebuild_upgrade(priority=1)
```
Comment (alastairong) [inline overlays/holo-nixpkgs/hpos-admin/hpos-admin.py:220]: Does this mean that /upgrade endpoint will work?

File: modules/system/holo-nixpkgs/auto-upgrade.nix [hunk 2]
```diff
@@ -16,15 +16,8 @@ in
   };
 
   config = mkIf cfg.enable {
-    systemd.services.holo-nixpkgs-auto-upgrade = {
```
Comment (alastairong) [inline modules/system/holo-nixpkgs/auto-upgrade.nix:19]: It looks like we're just moving the auto-upgrade itself to a separate nixos-rebuild service, and then calling it here. Can you explain a little more about what and why?

Comment (filalex77) [reply modules/system/holo-nixpkgs/auto-upgrade.nix:19]: Thanks for pointing out. I lost a changed `script` which calls `nix-channel --update` before rebuilding. That's what code reviews are for 😅 👀

File: overlays/holo-nixpkgs/hpos-admin/hpos-admin.py (hunk 1 above)
Comment (filalex77) [reply overlays/holo-nixpkgs/hpos-admin/hpos-admin.py:220]: Not yet, I need to resolve some newly appeared issues beforehand. After that, it should work.

File: modules/system/holo-nixpkgs/auto-upgrade.nix (hunk 2 above)
Comment (filalex77) [reply modules/system/holo-nixpkgs/auto-upgrade.nix:19]: Actually didn't lose it, misread the diff.

So this change is for sharing environment variables that Nix needs for `nixos-rebuild` between the two services, using `nixos-rebuild.service` as a base.

**Comment by filalex77** [comment]
Closing in favor of #555.

**Pull request closed without merging**